libraries/app/database_api.cpp

```diff
@@ -922,124 +925,124 @@ std::map<std::string, full_account> database_api_impl::get_full_accounts( const
          acnt.cashback_balance = account->cashback_balance(_db);
       }
 
-      uint64_t api_limit_get_full_accounts = _app_options->api_limit_get_full_accounts;
+      uint64_t api_limit_get_full_accounts_lists = _app_options->api_limit_get_full_accounts_lists;
 
       // Add the account's proposals
       auto  required_approvals_itr = proposals_by_account._account_to_proposals.find( account->id );
       if( required_approvals_itr != proposals_by_account._account_to_proposals.end() )
       {
-         acnt.proposals.reserve( std::min(required_approvals_itr->second.size(), api_limit_get_full_accounts) );
+         acnt.proposals.reserve( std::min(required_approvals_itr->second.size(), api_limit_get_full_accounts_lists) );
+         if(required_approvals_itr->second.size() > api_limit_get_full_accounts_lists)
+            acnt.more_data_available.proposals = true;
          for( auto proposal_id : required_approvals_itr->second )
          {
             acnt.proposals.push_back( proposal_id(_db) );
-            if(acnt.proposals.size() >= api_limit_get_full_accounts) {
-               acnt.more_data_available.proposals = true;
+            if(acnt.proposals.size() >= api_limit_get_full_accounts_lists)
                break;
-            }
          }
       }
 
       // Add the account's balances
       const auto& balances = _db.get_index_type< primary_index< account_balance_index > >().get_secondary_index< balances_by_account_index >().get_account_balances( account->id );
+      if(balances.size() > api_limit_get_full_accounts_lists)
+         acnt.more_data_available.balances = true;
       for( const auto balance : balances )
       {
          acnt.balances.emplace_back( *balance.second );
-         if(acnt.balances.size() >= api_limit_get_full_accounts) {
-            acnt.more_data_available.balances = true;
+         if(acnt.balances.size() >= api_limit_get_full_accounts_lists)
             break;
-         }
       }
 
       // Add the account's vesting balances
       auto vesting_range = _db.get_index_type<vesting_balance_index>().indices().get<by_account>().equal_range(account->id);
+      if((unsigned)abs(distance(vesting_range.first, vesting_range.second)) > api_limit_get_full_accounts_lists)
```

Comment:
According to https://en.cppreference.com/w/cpp/iterator/distance, complexity of `distance()` should be linear for boost multi-index container which is implemented with red-black tree. IMHO we should avoid `instance()` call when possible. Checking size before pushing a new item to the list is cheaper when the distance is large, also not more expensive when distance is small.
```
for( itr = start; itr != end; ++itr )
{
  if( list_size >= max )
   {
      has_more = true;
      break;
   }
   insert_to_list;
}
```

Reply:
ok, thank you for the detailed explication. i had used your code in all lists here https://github.com/bitshares/bitshares-core/pull/1749/commits/bda09a0989d8b5c030792b1dc6ecc12217b43868